core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/BaseDefaultHeartbeatPropertyProvider.java

```diff
@@ -0,0 +1,162 @@
+
+  /**
+   * Collection holding default properties for this default provider.
+   */
+  private final Set<String> defaultFields;
+
+  /**
+   * Random GUID that would help in analysis when app has stopped and restarted. Each restart will
+   * have a new GUID. If the application is unstable and goes through frequent restarts this will help
+   * us identify instability in the analytics backend.
+   */
+  private static UUID uniqueProcessId;
+
+  /**
+   * Name of this provider.
+   */
+  private final String name = "Base";
+
+  public BaseDefaultHeartbeatPropertyProvider() {
+    defaultFields = new HashSet<>();
+    initializeDefaultFields(defaultFields);
+  }
+
+  @Override
+  public String getName() {
+    return this.name;
+  }
+
+  @Override
+  public boolean isKeyWord(String keyword) {
+    return defaultFields.contains(keyword);
+  }
+
+  @Override
+  public Callable<Boolean> setDefaultPayload(final List<String> disableFields,
+      final HeartBeatProviderInterface provider) {
+    return new Callable<Boolean>() {
+
+      // using volatile here to avoid caching in threads.
+      volatile boolean hasSetValues = false;
+      volatile Set<String> enabledProperties = MiscUtils.except(defaultFields, disableFields);
+      @Override
+      public Boolean call() {
+        for (String fieldName : enabledProperties) {
+          try {
+            switch (fieldName) {
+              case "jdkVersion":
+                provider.addHeartBeatProperty(fieldName, getJdkVersion(), true);
+                hasSetValues = true;
+                break;
+              case "sdk-version":
+                provider.addHeartBeatProperty(fieldName, getSdkVersion(), true);
+                hasSetValues = true;
+                break;
+              case "osType":
+                provider.addHeartBeatProperty(fieldName, getOsType(), true);
+                hasSetValues = true;
+                break;
+              case "processSessionId":
+                provider.addHeartBeatProperty(fieldName, getProcessSessionId(), true);
+                hasSetValues = true;
+                break;
+              default:
+                //We won't accept unknown properties in default providers.
+                InternalLogger.INSTANCE.trace("Encountered unknown default property");
+                break;
+            }
+          }
+          catch (Exception e) {
+           InternalLogger.INSTANCE.warn("Failed to obtain heartbeat property, stack trace"
+               + "is: %s", ExceptionUtils.getStackTrace(e));
+          }
+        }
+        return hasSetValues;
+      }
+    };
+  }
+
+  /**
+   * This method initializes the collection with Default Properties of this provider.
+   * @param defaultFields collection to hold default properties.
+   */
+  private void initializeDefaultFields(Set<String> defaultFields) {
+
+    if (defaultFields == null) {
+      defaultFields = new HashSet<>();
+    }
+    defaultFields.add("jdkVersion");
```

Comment:
@cijothomas  Might be a good idea, but since there are different classes for each Different kind of providers I think doesn't add a lot of value since the separation of concerns is already addressed.

Reply:
It's generally good practice to refactor these as constants; especially if you have string "keys" that are consistent (especially if you use them in tests).

Reply:
@littleaj this is addressed